### DISCUSSION TITLE
tuning where preview text starts to include more complete query term highlighting

### DISCUSSION THREAD

**Discussion by designbymichael** [krisk/Fuse#740] [Q&A] [answered]
Question (or possibly feature request if this cannot be done):

So the `${snippet}` that it starts at the 0 mark of content, and I know with my content at least that it can be a fairly common occurrence that the phrase you're searching for is much further along in the content and often outside of reasonable snippet ranges.

Is there a way to tune the display of the snippet more, so that at least one occurrence of a matching phrase would be displayed?  For example:  if I'm searching with the query `Dark Side of the Moon`, I get lots of highlighted "of"s and "the"s, whereas the full term might be elsewhere on the page, and I may never show up in the snippet (which this can lead to a lack of confidence for the end-user that the page actually has that result.

**Comment by krisk** [comment, marked as answer]
Late reply — sorry about that.

Snippet generation and display is outside Fuse's scope — Fuse returns match indices, but how you render them is up to you. That said, the `matches[].indices` array gives you everything you need to build a smarter snippet.

For example, you could find the best match range (longest contiguous match) and center your snippet around that:

```js
const bestMatch = result.matches[0].indices
  .reduce((best, curr) => (curr[1] - curr[0] > best[1] - best[0]) ? curr : best)

const snippetStart = Math.max(0, bestMatch[0] - 40)
const snippetEnd = Math.min(text.length, bestMatch[1] + 40)
const snippet = text.slice(snippetStart, snippetEnd)
```

Also, `minMatchCharLength` can help filter out the noisy short matches ("of", "the") so your highlighting is cleaner.
